src/main/java/org/openrewrite/java/testing/junit5/MockitoRunnerToMockitoExtension.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junit5;
+
+import org.openrewrite.AutoConfigure;
+import org.openrewrite.CompositeRefactorVisitor;
+import org.openrewrite.Formatting;
+import org.openrewrite.Tree;
+import org.openrewrite.java.JavaIsoRefactorVisitor;
+import org.openrewrite.java.search.SemanticallyEqual;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.maven.AddDependency;
+import org.openrewrite.maven.MavenRefactorVisitor;
+import org.openrewrite.maven.tree.Maven;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.openrewrite.Formatting.EMPTY;
+import static org.openrewrite.Tree.randomId;
+import static org.openrewrite.java.testing.junit5.FrameworkTypes.extendWithIdent;
+import static org.openrewrite.java.testing.junit5.FrameworkTypes.extendWithType;
+import static org.openrewrite.java.testing.junit5.FrameworkTypes.runWithIdent;
+import static org.openrewrite.java.testing.junit5.FrameworkTypes.runWithType;
+
+/**
+ * Convert @RunWith(MockitoJUnitRunner.class) and @RunWith(MockitoJUnit44Runner.class) to @ExtendWith(MockitoExtension.class)
+ * MockitoJUnitRunner was part of the core library, but the MockitoExtension is in its own module.
+ * So this will attempt to add a dependency on mockito-junit-jupiter so that MockitoExtension can be on the classpath.
+ *
+ * This is part of the JUnit5 upgrade rather than the Mockito upgrade because you'd only need this if you were adopting JUnit5.
+ * Mockito can be upgraded from 1 to 3 without this on JUnit 4.
+ */
+@AutoConfigure
+public class MockitoRunnerToMockitoExtension extends CompositeRefactorVisitor {
+
+    public MockitoRunnerToMockitoExtension() {
+        AnnotationUpdate annotationUpdate = new AnnotationUpdate();
+        DependencyUpdate dependencyUpdate = new DependencyUpdate(annotationUpdate);
+        addVisitor(annotationUpdate);
+        addVisitor(dependencyUpdate);

```

Comment:
What is the difference between `addVisitor()` vs doing `andThen`? Do we need to do this because this is a CompositeRefactorVisitor? If we could conditionally add the depdenecyUpdate visitor via `andThen()`, we could get away from having a flag to coordinate the behavior between the two visitors; The first visitor would only add the second visitor if we know we are going to need the dependency update.

Reply:
There is very, very little difference. Indeed, `CompositeRefactorVisitor` andThen()s the visitors you pass to it.
Looking through the code the only reason you _can't_ andThen() visitors of other types is a type constraint we self-imposed. I'm not sure if that constraint actually provides any value... Worth considering, but out of scope for this PR.